x/liquidity/legacy/v3/store.go

```diff
@@ -0,0 +1,60 @@
+package v3
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	v2liquidity "github.com/cosmosquad-labs/squad/v2/x/liquidity/legacy/v2"
+	"github.com/cosmosquad-labs/squad/v2/x/liquidity/types"
+)
+
+func MigrateOrders(store sdk.KVStore, cdc codec.BinaryCodec) error {
+	iter := sdk.KVStorePrefixIterator(store, types.OrderKeyPrefix)
+	defer iter.Close()
+
+	for ; iter.Valid(); iter.Next() {
+		var oldOrder v2liquidity.Order
+		if err := cdc.Unmarshal(iter.Value(), &oldOrder); err != nil {
+			return err
+		}
+
+		newOrder := types.Order{
+			// There's no way to determine whether the order was made through
+			// MsgLimitOrder or MsgMarketOrder, set the order type as OrderTypeLimit
+			// as a fallback.
+			Type:      types.OrderTypeLimit,
```

Comment:
Am I understanding correctly that this migrates all the previous orders to `OrderTypeLimit`? I don't know if this will cause any issue.

Reply:
Orders stay for at most one day and since this `Type` field is just for additional information, marking past orders made through `MsgMarketOrder` as `OrderTypeLimit` will not cause any issue AFAIK. After the migration, new market orders will be correctly marked as `OrderTypeMarket`.